ConsentViewController/Classes/SourcePointClient/Adapters.swift

```diff
@@ -108,6 +108,57 @@ extension SPMobileCore.ConsentStatus.ConsentStatusGranularStatus {
     }
 }
 
+extension Bool? {
+    func toCore() -> KotlinBoolean? {
+        // swiftlint:disable:next force_unwrapping
+        return self != nil ? KotlinBoolean(value: self!) : nil
+    }
+}
+
+extension ConsentStatus {
+    func toCore(rejectedVendors: [String]? = nil, rejectedCategories: [String]? = nil) -> SPMobileCore.ConsentStatus {
+        return SPMobileCore.ConsentStatus.init(
+            rejectedAny: rejectedAny.toCore(),
+            rejectedLI: rejectedLI.toCore(),
+            rejectedAll: rejectedAll.toCore(),
+            consentedAll: consentedAll.toCore(),
+            consentedToAll: consentedToAll.toCore(),
+            consentedToAny: consentedToAny.toCore(),
+            hasConsentData: hasConsentData.toCore(),
+            vendorListAdditions: vendorListAdditions.toCore(),
+            legalBasisChanges: legalBasisChanges.toCore(),
+            granularStatus: granularStatus?.toCore(),
+            // swiftlint:disable:next force_unwrapping
+            rejectedVendors: rejectedVendors != nil ? rejectedVendors! as [String] : [String](),
+            // swiftlint:disable:next force_unwrapping
+            rejectedCategories: rejectedVendors != nil ? rejectedCategories! as [String] : [String]()
+        )
+    }
+}
```

Comment:
I had to implement the `ConsentStatus.toCore()` function, but I chose to extend `KotlinBoolean` instead of `Bool`. That way we don't have to rely on force unwraps.
```swift
extension KotlinBoolean {
    public convenience init?(bool value: Bool?) {
        guard let bool = value else { return nil }
        self.init(bool: bool)
    }
}

extension ConsentStatus {
    func toCore() -> SPMobileCore.ConsentStatus {
        return SPMobileCore.ConsentStatus(
            rejectedAny: KotlinBoolean(bool: rejectedAny),
            rejectedLI: KotlinBoolean(bool: rejectedLI),
            rejectedAll: KotlinBoolean(bool: rejectedAll),
            consentedAll: KotlinBoolean(bool: consentedAll),
            consentedToAll: KotlinBoolean(bool: consentedToAll),
            consentedToAny: KotlinBoolean(bool: consentedToAny),
            hasConsentData: KotlinBoolean(bool: hasConsentData),
            vendorListAdditions: KotlinBoolean(bool: vendorListAdditions),
            legalBasisChanges: KotlinBoolean(bool: legalBasisChanges),
            granularStatus: SPMobileCore.ConsentStatus.ConsentStatusGranularStatus(
                vendorConsent: granularStatus?.vendorConsent,
                vendorLegInt: granularStatus?.vendorLegInt,
                purposeConsent: granularStatus?.purposeConsent,
                purposeLegInt: granularStatus?.purposeLegInt,
                previousOptInAll: KotlinBoolean(bool: granularStatus?.previousOptInAll),
                defaultConsent: KotlinBoolean(bool: granularStatus?.defaultConsent),
                sellStatus: KotlinBoolean(bool: granularStatus?.sellStatus),
                shareStatus: KotlinBoolean(bool: granularStatus?.shareStatus),
                sensitiveDataStatus: KotlinBoolean(bool: granularStatus?.sensitiveDataStatus),
                gpcStatus: KotlinBoolean(bool: granularStatus?.gpcStatus)
            )
        )
    }
}
```
ps. it's missing the two new attributes `rejectedVendors` and `rejectedCategories`. Which, btw, don't need to be unwrapped either, since their values are optionals, we can assign them as `rejectedVendors: rejectedVendors`

Reply:
@andresilveirah regarding `rejectedVendors` and `rejectedCategories` attributes: if I do not unwrap them like this
```
            // swiftlint:disable:next force_unwrapping
            rejectedVendors: rejectedVendors != nil ? rejectedVendors! as [String] : [String](),
            // swiftlint:disable:next force_unwrapping
            rejectedCategories: rejectedVendors != nil ? rejectedCategories! as [String] : [String]()
```
Then these attributes are missing in KMM's request body. 
Let me think of a correct way of handling these things. Maybe I'll introduce `KotlinArray` or something like this.